tests/samples_smoke_tests/common/samples_common_test_clas.py

```diff
@@ -0,0 +1,432 @@
+            if not is_perf:
+                # Skipping all tests for this sample, because no of them are ready for performance.
+                # Add name of sample to global pytest variable, then skip it in setup method
+                if 'list_of_skipped_samples' in Environment.env:
+                    Environment.env['list_of_skipped_samples'].append(self.sample_name)
+                else:
+                    Environment.env.update({'list_of_skipped_samples': [self.sample_name]})
+                pytest.skip('[INFO] Sample {} not executed for performance'.format(self.executable_path))
+            else:
+                log.info('Running perfomance for {} iteraction'.format(perf_iter))
+                # Perf_iter = 0 when it isn't neccessary to add niter key
+                if perf_iter > 0:
+                    if is_niter:
+                        log.warning('Changed value of niter param to {}'.format(perf_iter))
+                        param_cp['niter'] = perf_iter
+                    else:
+                        log.warning('Added key: niter to param with value: {}'.format(perf_iter))
+                        param_cp.update({'niter': perf_iter})
+                cmd_perf = get_cmd_func(param_cp, use_preffix=use_preffix, long_hyphen=long_hyphen)
+                retcode_perf, stdout_perf, stderr_perf = shell([self.executable_path, cmd_perf])
+                if (retcode_perf != 0):
+                    log.error(stderr_perf)
+                assert retcode_perf == 0, "Execution sample for performace failed"
+                fps_perf = self.find_fps(retcode_perf, stdout_perf, stderr_perf)
+                self.write_csv(sample_name=self.sample_name, sample_type=sample_type, cmd_perf=cmd_perf, fps_perf=fps_perf)
+                log.info('Perf results: {}'.format(fps_perf))
+        if get_shell_result:
+            return retcode, stdout, stderr
+        # Check return code
+        if (retcode != 0):
+            log.error(stderr)
+        assert retcode == 0, "Sample execution failed"     
+        return stdout
+
+    def setup_method(self):
+        """
+        Clean up IRs and npy files from self.output_dir if exist
+        And skip several test for performance
+        :return: """
+        if os.path.exists(self.output_dir):
+            shutil.rmtree(self.output_dir)
+        filenames = glob.glob('out*.bmp')
+        [os.remove(fn) for fn in filenames]
+        # Skip samples that are not for performance:
+        if Environment.env['performance'] and 'list_of_skipped_samples' in Environment.env and \
+                self.sample_name in Environment.env['list_of_skipped_samples']:
+            pytest.skip('[Skip from setup] Sample {} not executed for performance'.format(self.sample_name))
+
+    def teardown_method(self):
+        """
+        Clean up IRs and npy files from self.output_dir if exist
+        :return: """
+        if not hasattr(self, 'save') or not self.save:
```

Comment:
replace with `getattr` to check var exists and check it's value

Reply:
was replaced.